fuzzware_pipeline/__init__.py

```diff
@@ -890,15 +905,37 @@ def do_genstats(args, leftover_args):
         for main_dir in main_dirs_for_proj(projdir):
             logger.info(f"Crash contexts from main dir: {main_dir}")
             config_path = None
-
+            results_dict = {}
+            args_tuple_list = []
+            m = mp.Manager()
+            results_queue = m.Queue()
+            iterator = 0
+            num_processed = 0
             for crashing_input in nc.crash_paths_for_main_dir(main_dir):
+                # continue
                 if config_path is None:
                     config_path = config_for_input_path(crashing_input)
                     extra_args_file = extra_args_for_config_path(config_path)
                     extra_args = parse_extra_args(load_extra_args(extra_args_file), projdir)
                     if "-v" not in extra_args:
                         extra_args += ["-v"]
-                emu_output = str(run_target(config_path, crashing_input, extra_args, get_output=True, silent=True))
+                # emu_output = str(run_target(config_path, crashing_input, extra_args, get_output=True, silent=True))
+                args_tuple_list.append([config_path, crashing_input, extra_args, True, True, results_queue])
+                iterator += 1
+
+            p = mp.Process(target=multi_run_target_manager, args=(args_tuple_list,))
```

Comment:
I think mp.Process does not limit the number of concurrent processes. We want to be able to limit this number so that other fuzzing runs are not impacted by the trace generation taking up all computation.

`multiprocessing.Pool` should be able to limit this number.

Given that, we can add a `-n` argument similar to the `fuzzware pipeline -n <num_cores>` argument which allows setting the max number of cores/parallel processes used by trace generation.